### PR TITLE
semiphemeral: init at 0.6

### PR DESCRIPTION
**Pull request by amanjeev**
###### Motivation for this change

Adds python package [Semiphemeral](https://github.com/micahflee/semiphemeral). 

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 120619` at 34ebeb4f run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>python38Packages.semiphemeral</li>
    <li>python39Packages.semiphemeral</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/semiphemeral/default.nix:26:0:
```
   |
26 |     description = "Automatically delete your old tweets, except for the ones you want to keep";
   | ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/semiphemeral/default.nix:31]: add a final `\n`

File: pkgs/development/python-modules/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/semiphemeral/default.nix:11]:
```suggestion
buildPythonApplication rec {
```

File: pkgs/development/python-modules/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/semiphemeral/default.nix:15]:
```suggestion
  src = fetchPypi {
```

File: pkgs/development/python-modules/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/development/python-modules/semiphemeral/default.nix:22]:
```suggestion
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
This isn't a Python module (no `import semiphemeral`), so put it in `pkgs/tools/...`.

Also, you need exactly two commits:
* `maintainers: add amanjeev` and
* `semiphemeral: init at 0.6`

File: pkgs/tools/social/twitter/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/tools/social/twitter/semiphemeral/default.nix:9]:
```suggestion
{ lib
, python3
}:
```

File: pkgs/tools/social/twitter/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/tools/social/twitter/semiphemeral/default.nix:11]:
```suggestion
python3.pkgs.buildPythonApplication rec {
```

File: pkgs/tools/social/twitter/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/tools/social/twitter/semiphemeral/default.nix:13]:
```suggestion
  pname = "semiphemeral";
  version = "0.6";
```

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please also address @r-rmcgibbo's suggestion.

File: pkgs/tools/social/twitter/semiphemeral/default.nix
Comment (amanjeev) [reply pkgs/tools/social/twitter/semiphemeral/default.nix:9]: thank you, this is really a lot of experience building

Comment (dotlambda) [reply pkgs/tools/social/twitter/semiphemeral/default.nix:9]: The reason we do this is so that it's easier to override Python packages. That's done by using
```nix
semiphemeral.override {
  python3 = python3.override {
    packagesOverrides = ...;
  };
}
```

Comment (amanjeev) [reply pkgs/tools/social/twitter/semiphemeral/default.nix:9]: I am not sure if I understand.  This does not build. 

```
error: undefined variable 'click' at /home/aj/proj/nix-nixos/nixpkgs/pkgs/tools/social/twitter/semiphemeral/default.nix:16:29
```

File: pkgs/tools/social/twitter/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/tools/social/twitter/semiphemeral/default.nix:22]:
```suggestion
  propagatedBuildInputs = with python3.pkgs; [ click sqlalchemy flask tweepy colorama ];
```

File: pkgs/tools/social/twitter/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/tools/social/twitter/semiphemeral/default.nix:15]:
```suggestion
  src = python3.pkgs.fetchPypi {
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/tools/social/twitter/semiphemeral/default.nix
Comment (dotlambda) [reply pkgs/tools/social/twitter/semiphemeral/default.nix:9]: See the newest review for how to do it.

File: pkgs/top-level/all-packages.nix
Comment (amanjeev) [inline pkgs/top-level/all-packages.nix:25806]: I had to remove these too! Otherwise you get the error 

```
error: anonymous function at /home/aj/proj/nix-nixos/nixpkgs/pkgs/tools/social/twitter/semiphemeral/default.nix:1:1 called with unexpected argument 'buildPythonApplication', at /home/aj/proj/nix-nixos/nixpkgs/lib/customisation.nix:69:16
```

Comment (dotlambda) [reply pkgs/top-level/all-packages.nix]: Yes, you do have to remove them.

**Comment by dotlambda** [comment]
Mind the two commits.

File: pkgs/top-level/all-packages.nix
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix:25800]:
```suggestion
  semiphemeral = callPackage ../tools/social/twitter/semiphemeral { };
```

**Comment by amanjeev** [comment]
Would it be ok to squash the commits before merging or should I squash and push again?

**Review by dotlambda — CHANGES_REQUESTED** [review]
You should set `doCheck = false` and add a comment since upstream seems to have no tests.

**Comment by dotlambda** [comment]
> Would it be ok to squash the commits before merging or should I squash and push again?

I can only squash multiple commits into a single one when merging.

**Comment by amanjeev** [comment]
> Mind the two commits.

Yea I am can squash n -1 on my side and push again once done?

File: pkgs/top-level/all-packages.nix
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix:25798]: Sorry I didn't see this before.
```suggestion
  semiphemeral = callPackage ../tools/misc/semiphemeral { };
```

**Review by dotlambda — APPROVED** [review]
LGTM except for the path.

**Comment by amanjeev** [comment]
thank you so much @dotlambda for your patience with me

**Review by Hoverbear — APPROVED** [review]
Tested on NixOS unstable and works fine.

File: pkgs/tools/misc/semiphemeral/default.nix
Comment (dotlambda) [inline pkgs/tools/misc/semiphemeral/default.nix:19]:
```suggestion

```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by amanjeev** [comment]
@Hoverbear suggested I use `nixpjgs-fmt` before committing. Will do and push! Sorry about this.

**Pull request merged**
